guppylang/tys/builtin.py

```diff
@@ -35,28 +33,10 @@ class _CallableTypeDef(TypeDef):
     name: Literal["Callable"] = field(default="Callable", init=False)
 
     def check_instantiate(
-        self, args: FlaggedArgs, globals: "Globals", loc: AstNode | None = None
+        self, args: Sequence[Argument], globals: "Globals", loc: AstNode | None = None
     ) -> FunctionType:
-        # We get the inputs/output as a flattened list: `args = [*inputs, output]`.
-        if not args:
-            raise GuppyError(f"Missing parameter for type `{self.name}`", loc)
-        args = [
-            # TODO: Better error location
-            (TypeParam(0, f"T{i}", can_be_linear=True).check_arg(arg, loc).ty, flags)
-            for i, (arg, flags) in enumerate(args)
-        ]
-        *inputs, (output_ty, output_flags) = args
-        for ty, flags in inputs:
-            if InputFlags.Inout in flags and not ty.linear:
-                raise GuppyError(
-                    f"Non-linear type `{ty}` cannot be annotated as `@inout`",
-                    loc,  # TODO: Better error location
-                )
-        if output_flags != InputFlags.NoFlags:
-            raise GuppyError(
-                "`@` type annotations are not allowed in this position", loc
-            )
-        return FunctionType([FuncInput(ty, flags) for ty, flags in inputs], output_ty)
+        # Callable types are constructed using special login in the type parser
```

Comment:
```suggestion
        # Callable types are constructed using special logic in the type parser
```

Reply:
You could consider moving some of that code into CallableTypeDef itself, just so you could have a comment "Callable types are constructed using instantiate_func_type()" or something like that, but this is fine